frontend/src/v5/ui/routes/viewer/tickets/commentsPanel/comment/comment.helpers.ts

```diff
@@ -0,0 +1,90 @@
+import { CommentReplyMetadata, IComment } from '@/v5/store/tickets/tickets.types';
+import _ from 'lodash';
+
+// Message metadata and text functions
+export const MAX_MESSAGE_LENGTH = 1200;
+export const deletedCommentMessage = formatMessage({ id: 'ticket.comment.message.deleted', defaultMessage: 'Message deleted' });
+export const editedCommentMessage = formatMessage({ id: 'ticket.comment.message.edited', defaultMessage: 'edited' });
+
+const extractMetadataValue = (message: string, metadataName: keyof CommentReplyMetadata) => {
+	// eslint-disable-next-line security/detect-non-literal-regexp
+	const regex = new RegExp(`\\[${metadataName}\\]:- "([^"]*)"\\n`);
+	return regex.exec(message)?.[1] || '';
+};
+
+export const extractMetadata = (message: string): CommentReplyMetadata => ({
+	_id: extractMetadataValue(message, '_id'),
+	author: extractMetadataValue(message, 'author'),
+	message: extractMetadataValue(message, 'message'),
+});
+
+export const createMetadata = (comment: IComment): CommentReplyMetadata => (
+	_.pick(comment, '_id', 'author', 'message') as CommentReplyMetadata
+);
+
+export const extractMessage = (message: string) => message.replaceAll(/\[[_a-z]*\]:- ".*"\n[\n]?/g, '');
+export const sanitiseMessage = (message: string) => message.replaceAll('"', '&#34;').replaceAll('\n', '<br />');
+export const desanitiseMessage = (message: string) => message.replaceAll('&#34;', '"').replaceAll('<br />', '\n');
+
+const createMetadataValue = (metadataName: keyof CommentReplyMetadata, metadataValue: string) => (
+	`[${metadataName}]:- "${metadataValue}"\n`
+);
+
+const stringifyMetadata = ({ author, _id, message }: CommentReplyMetadata) => {
+	const metadata = [
+		createMetadataValue('_id', _id),
+		createMetadataValue('author', author),
+		createMetadataValue('message', extractMessage(message)),
+	];
+	return metadata.join('');
+};
+
+export const addReply = (metadata: CommentReplyMetadata, newMessage: string) => `${stringifyMetadata(metadata)}\n${newMessage}`;
+
+// Time related functions
+const TIME_UNIT = {
+	second: formatMessage({ id: 'timeUnit.second', defaultMessage: 'second' }),
+	minute: formatMessage({ id: 'timeUnit.minute', defaultMessage: 'minute' }),
+	hour: formatMessage({ id: 'timeUnit.hour', defaultMessage: 'hour' }),
+	day: formatMessage({ id: 'timeUnit.day', defaultMessage: 'day' }),
+	month: formatMessage({ id: 'timeUnit.month', defaultMessage: 'month' }),
+	year: formatMessage({ id: 'timeUnit.year', defaultMessage: 'year' }),
+};
+
+export const getRelativeTime = (from: Date) => {
+	let timeDifference = (new Date().getTime() - from.getTime()) / 1000;
+	if (timeDifference < 60) return formatRelativeTime(-Math.ceil(timeDifference), TIME_UNIT.second);
+
+	timeDifference /= 60;
+	if (timeDifference < 60) return formatRelativeTime(-Math.ceil(timeDifference), TIME_UNIT.minute);
+
+	timeDifference /= 60;
+	if (timeDifference < 24) return formatRelativeTime(-Math.ceil(timeDifference), TIME_UNIT.hour);
```

Comment:
I'm unsure about this getRelativeTime function. It seems to jump straight from "59 minutes" to "2 hours"

![CommentTime](https://user-images.githubusercontent.com/62481250/216954594-454fa233-2e65-416b-adc3-23cb6f5dbd51.png)
(Note it only says 1 hour when it is exactly 1 hour old in ms then rounds up to 2 hours)
Are there any problems to using `round` rather than `ceil`?

Reply:
I was using `ceil` because when posting the first message, neither `floor` nor `round` would produce a good result (`0 seconds`). Then I just applied it to all the other times without realising about this issue that you raised.

I am now using `floor` to all of them, but not ceil because I am not sure 15 days ago should display as `1 month ago`, is that the right behaviour @mosment3DRepo ?

Reply:
Surely the floor/round/ceil only affects the displayed number and not the unit?

15 days will still trigger the `if (timeDifference < 30)` below and use the unit 'Days'. it's just that the value will be either 15 or 16.

Reply:
you're right, sorry, I meant 1 month and 15 days would display as 2 months.

Reply:
you're right, sorry, I meant 1 month and 15 days would display as 2 months.